include/api.php

```diff
@@ -114,12 +114,12 @@ function pll_default_language( $field = 'slug' ) {
  *
  * @param int                 $post_id Post ID.
  * @param PLL_Language|string $lang    Optional language (object or slug), defaults to the current language.
- * @return int|false The translation post ID if exists, otherwise the passed ID. False if the passed object has no language or if the language doesn't exist.
+ * @return int|false The translation post ID if exists, 0 if not translated or if the post has no language. False if the language doesn't exist.
  *
  * @phpstan-return int<0, max>|false
  */
-function pll_get_post( $post_id, $lang = '' ) {
-	$lang = $lang ? $lang : pll_current_language();
+function pll_get_post( $post_id, $lang = null ) {
+	$lang = $lang ? PLL()->model->get_language( $lang ) : pll_current_language();
```

Comment:
I don't like this because you are mixing 2 types. Moreover `PLL_Translated_Object::get()` make the conversion anyway. So I propose to only check if a language was provided or not?
```suggestion
	if ( empty( $lang ) ) {
		$lang = pll_current_language();
	}
```
The second check after this is probably useless as included in `PLL_Translated_Object::get()`.

Reply:
Well I reverted changes on this line in 6ff20f1d.
No sure what you suggest is correct since `PLL_Translated_Object::get()` doesn't accept `false` and `pll_current_language()` *can* return `false`. What do you think?